From: Balint Reczey <balint.reczey@canonical.com>
Date: Sat, 3 Apr 2021 13:23:28 +0200
Subject: Revert "pager: stop disabling urlification under a pager"

Debian and Ubuntu does not yet have a less version that supports urlification
https://github.com/systemd/systemd/issues/18814

This patch can be dropped when less 563 or later enters the archive

This reverts commit ebef02dd8f35f8b6ee58060b71b7f321a5027760.
---
 src/shared/pretty-print.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/shared/pretty-print.c b/src/shared/pretty-print.c
index 26daec3..2c742a0 100644
--- a/src/shared/pretty-print.c
+++ b/src/shared/pretty-print.c
@@ -22,6 +22,10 @@ bool urlify_enabled(void) {
 #if ENABLE_URLIFY
         static int cached_urlify_enabled = -1;
 
+        /* Unfortunately 'less' doesn't support links like this yet ðŸ˜­, hence let's disable this as long as there's a
+         * pager in effect. Let's drop this check as soon as less got fixed a and enough time passed so that it's safe
+         * to assume that a link-enabled 'less' version has hit most installations. */
+
         if (cached_urlify_enabled < 0) {
                 int val;
 
@@ -29,7 +33,7 @@ bool urlify_enabled(void) {
                 if (val >= 0)
                         cached_urlify_enabled = val;
                 else
-                        cached_urlify_enabled = colors_enabled();
+                        cached_urlify_enabled = colors_enabled() && !pager_have();
         }
 
         return cached_urlify_enabled;
